hack/lib/version.sh

```diff
@@ -50,6 +50,13 @@ kube::version::get_version_vars() {
 
     # Use git describe to find the version based on annotated tags.
     if [[ -n ${KUBE_GIT_VERSION-} ]] || KUBE_GIT_VERSION=$("${git[@]}" describe --tags --abbrev=14 "${KUBE_GIT_COMMIT}^{commit}" 2>/dev/null); then
+      # This translates the "git describe" to an actual semver.org
+      # compatible semantic version that looks something like this:
+      #   v1.1.0-alpha.0.6+84c76d1142ea4d
+      #
+      # TODO: We continue calling this "git version" because so many
+      # downstream consumers are expecting it there.
+      KUBE_GIT_VERSION=$(echo "${KUBE_GIT_VERSION}" | sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{14\}\)$/.\1\+\2/")
```

Comment:
I think we want: sed "s/-\([0-9]\{1,\}\)-g\([0-9a-f]\{14\}\)$/-\1\+\2/")

The semver line 9 says that you may include `-` inside the prerelease information. On the release branch this would mean you get versions like `v1.0.2-2+abcdef`  which is semver happy.  On the master branch this would mean get versions like `v1.2.0-alpha.0-1+abcdef` which is also semver happy.